gobblin-core/src/main/java/gobblin/writer/commands/TeradataBufferedInserter.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2016 Swisscom All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.writer.commands;
+
+import gobblin.configuration.State;
+import gobblin.converter.jdbc.JdbcEntryData;
+import gobblin.converter.jdbc.JdbcEntryDatum;
+
+import java.sql.BatchUpdateException;
+import java.sql.Connection;
+import java.sql.ParameterMetaData;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import lombok.ToString;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+
+/**
+ * The implementation of JdbcBufferedInserter for Teradata.
+ * Writing is done by executing {@link JdbcBufferedInserter#WRITER_JDBC_INSERT_BATCH_SIZE} sized batch inserts.
+ * 
+ * @author Lorand Bendig
+ *
+ */
+@ToString
+public class TeradataBufferedInserter implements JdbcBufferedInserter {
+  private static final Logger LOG = LoggerFactory.getLogger(TeradataBufferedInserter.class);
+
+  private static final String INSERT_STATEMENT_PREFIX_FORMAT = "INSERT INTO %s.%s (%s) VALUES ";
+  private static final Joiner JOINER_ON_COMMA = Joiner.on(',');
+
+  private List<JdbcEntryData> pendingInserts;
```

Comment:
For this I restructured the *`BufferedInserter` classes. Because some of the methods were just copy pasted from `MySQLBufferedInserter` to the Teradata one, I refactored the
code a bit and now these are in an abstract `BaseJdbcBufferedInserter` class.